lib/internal/modules/esm/get_format.js

```diff
@@ -51,7 +52,8 @@ function defaultGetFormat(url, context, defaultGetFormatUnused) {
     const ext = extname(parsed.pathname);
     let format;
     if (ext === '.js') {
-      format = getPackageType(parsed.href) === 'module' ? 'module' : 'commonjs';
+      const type = getPackageType(parsed.href);
+      format = type !== 'none' ? type : defaultPackageType;
```

Comment:
This `defaultPackageType` implementation can move into the `getPackageType` implementation itself. Alternatively `getPackageType` can return eg `undefined` to indicate the default should be used.

I just want to avoid explicitly supporting a new `"type": "none"` value in the public API.